### PR TITLE
Submit a remote job which needs a sessionId to retrieve

### PR DESCRIPTION
**Pull request by tobwiens**
Problem:
- The secure catalog requests a sessionId to access workflows

Solution:
- Add a new endpoint which allows to retrieve workflows from endpoints which are secured with a sessionId

### CODE REVIEW

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java [hunk 1]
```diff
@@ -126,6 +127,26 @@ JobId submit(URL job)
      * precious result.
      * <p>
      *
+     * @param job       a job provided as a url
+     * @param variables job variables to use during the job execution
```
Comment (fviale) [inline rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java]: please add the headerParams in the javadoc

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java [hunk 2]
```diff
@@ -705,19 +707,29 @@ public JobId submit(File job)
...
         JobIdData jobIdData = null;
         try {
-            InputStream is = job.openStream();
-            jobIdData = restApiClient().submitXml(sid, is);
+            URLConnection urlConnection = job.openConnection();
+
+            for(Map.Entry<String,String> requestHeaderEntry: requestHeaderParams.entrySet()) {
+                urlConnection.addRequestProperty(requestHeaderEntry.getKey(), requestHeaderEntry.getValue());
+            }
+            InputStream is = urlConnection.getInputStream();
```
Comment (fviale) [inline rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java]: don't you need to call urlConnection.connect() before ?

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 232 issues
* MAJOR 3 major
* MINOR 229 minor

#### Top 10 issues

1. MAJOR [SchedulerClient.java#L729](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L729): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MAJOR [SchedulerClient.java#L729](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L729): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MAJOR [SchedulerClient.java#L748](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L748): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MINOR [SchedulerNodeClient.java#L117](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L117): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L124](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L124): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L130](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L130): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L136](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L136): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L142](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L142): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L149](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L149): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L156](https://github.com/ow2-proactive/scheduling/blob/04d70ee58e53bfc6dc2be7cf6cc1dc79e220d4c2/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L156): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java (hunk 1 above)
Comment (tobwiens) [reply rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java]: Good catch, fixed

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 231 issues
* MAJOR 3 major
* MINOR 228 minor

#### Top 10 issues

1. MAJOR [SchedulerClient.java#L730](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L730): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MAJOR [SchedulerClient.java#L730](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L730): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MAJOR [SchedulerClient.java#L749](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L749): Replace "Collections.EMPTY_MAP" by "Collections.emptyMap()". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1596)
1. MINOR [SchedulerNodeClient.java#L117](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L117): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L124](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L124): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L130](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L130): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L136](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L136): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L142](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L142): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L149](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L149): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L156](https://github.com/ow2-proactive/scheduling/blob/ce17c50d3682505b4aa98ccfdab68b1082155282/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L156): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by tobwiens** [comment]
Retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 228 issues
* MINOR 228 minor

#### Top 10 issues

1. MINOR [SchedulerNodeClient.java#L117](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L117): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L124](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L124): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L130](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L130): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L136](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L136): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L142](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L142): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L149](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L149): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L156](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L156): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L163](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L163): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L170](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L170): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L176](https://github.com/ow2-proactive/scheduling/blob/b479bde9b30ccad9f138e8d899113c3a7f2f5c47/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L176): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 228 issues
* MINOR 228 minor

#### Top 10 issues

1. MINOR [SchedulerNodeClient.java#L117](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L117): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L124](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L124): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L130](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L130): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L136](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L136): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L142](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L142): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L149](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L149): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L156](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L156): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L163](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L163): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L170](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L170): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerNodeClient.java#L176](https://github.com/ow2-proactive/scheduling/blob/f8e9f3bc77c6d3cc16f74e6d909a9c48ef46228a/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java#L176): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by tobwiens** [comment]
Retest this please

**Pull request merged**